target/scripts/startup/setup.d/security/rspamd.sh

```diff
@@ -304,6 +305,56 @@ function __rspamd__setup_check_authenticated() {
   fi
 }
 
+# If `RSPAMD_SPAM_SUBJECT` is not empty, we create a Sieve script that alters the `Subject`
+# header, in order to prepend a user-defined string.
+function __rspamd__setup_spam_subject() {
+  if [[ -z ${RSPAMD_SPAM_SUBJECT} ]]
+  then
+    __rspamd__log 'debug' 'Spam subject is not set - no prefix will be added to spam e-mails'
+  else
+    __rspamd__log 'debug' "Spam subject is set - the prefix '${RSPAMD_SPAM_SUBJECT}' will be added to spam e-mails"
+
+    __rspamd__log 'trace' "Enabling '+editheader' Sieve extension"
+    # check whether sieve_global_extensions is disabled (and enabled it if so)
+    sed -i -E 's|#(sieve_global_extensions.*)|\1|' /etc/dovecot/conf.d/90-sieve.conf
+    # then append the extension
+    sedfile -i -E 's|(sieve_global_extensions.*)|\1 +editheader|' /etc/dovecot/conf.d/90-sieve.conf
+
+    # This directory contains Sieve scripts that are executed before user-defined Sieve
+    # scripts run.
+    local DOVECOT_SIEVE_GLOBAL_BEFORE_DIR='/usr/lib/dovecot/sieve-global/before'
+    local DOVECOT_SIEVE_FILE='rspamd_spam_subject'
+    readonly DOVECOT_SIEVE_GLOBAL_BEFORE_DIR DOVECOT_SIEVE_FILE
+
+    mkdir -p "${DOVECOT_SIEVE_GLOBAL_BEFORE_DIR}"
+    # ref: https://superuser.com/a/1502589
+    cat >"${DOVECOT_SIEVE_GLOBAL_BEFORE_DIR}/${DOVECOT_SIEVE_FILE}.sieve" << EOF
+require ["editheader","variables"];
+
+if header :contains "X-Spam" "Yes"
```

Comment:
could it also handle `X-Spam-Flag` ?
https://is.oregonstate.edu/exchange-email-services/spamassassin

I am not sure this is right or not
I have this file that adds this header: https://github.com/wdes/mails.wdes.eu/blob/f40c1fec5a67b3448df4a536bf790d1ae215ec57/rspamd/local.d/milter_headers.conf#L16

Reply:
We could easily add something like `anyof (header :contains ...` to match multiple headers. The question is: how many do we want to support? One idea would be to have a new ENV like `RSPAMD_SPAM_HEADER_NAME` so we have a uniform name.

Reply:
Looks like a good idea to have `RSPAMD_SPAM_HEADER_NAME`

Also, the value may vary as I have observed on other repos, like https://github.com/mailcow/mailcow-dockerized/blob/2024-01/data/conf/dovecot/global_sieve_after#L11

Sometimes it is uppercase, some other times not. Depending on the example/tutorial/app that you are reading

Reply:
> PS: move to junk does not work at all for me for unknown reasons, maybe this integration will make it work better

It probably does not work because of this:

https://github.com/docker-mailserver/docker-mailserver/blob/00018e7e2b226fafb3a5ab50db078885ca6c8ddc/target/scripts/startup/setup.d/security/misc.sh#L249-L270

It's the same story as with this PR I guess.

---

Having `RSPAMD_SPAM_HEADER(_NAME)` is something I can implement in a follow-up PR.

Reply:
I concur, having too many ENVs is problematic. Let's wait with tihs.